### DISCUSSION TITLE
Is FifoBuffer thread safe

### DISCUSSION THREAD

**Discussion by xzhan96** [google/oboe#1402] [Q&A] [answered]
I read the implementation of FifoBuffer, and see the read/write pos is protected by atomic, if it means the Fifobuffer is thread safe?
thanks a lot for your kindly answer.

**Comment by dturner** [comment, marked as answer]
Looking at [the code](https://github.com/google/oboe/blob/4fa162bc5f1949f4130e1ee43ae237e8359b8d9c/src/fifo/FifoBuffer.cpp#L69) it looks like `FifoBuffer` is thread-safe for single-reader and single-writer only. If you have more than one reader or writer thread it'll break.
